### PR TITLE
New Stats graphs - APPM-1092

### PR DESCRIPTION
**Pull request by lakshani**
This includes the fixes for APPM-1085,APPM-1086,APPM-1092,APPM-1093

### CODE REVIEW

File: components/org.wso2.carbon.appmgt.hostobjects/src/main/java/org/wso2/carbon/appmgt/hostobjects/APIProviderHostObject.java [hunk 1]
```diff
@@ -3487,6 +3487,68 @@ public static NativeArray jsFunction_getAppsByEndPoint(Context cx, Scriptable th
...
+            String tenantDomain = MultitenantUtils.getTenantDomain(
+                    AppManagerUtil.replaceEmailDomainBack(providerName));
+            if(tenantDomain != null && !MultitenantConstants.SUPER_TENANT_DOMAIN_NAME.equals(
+                    tenantDomain)) {
+                isTenantFlowStarted = true;
+                PrivilegedCarbonContext.startTenantFlow();
+                PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain(
+                        tenantDomain, true);
+            }
+            int tenantId = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantId(true);
```
Comment (lakshani) [inline components/org.wso2.carbon.appmgt.hostobjects/src/main/java/org/wso2/carbon/appmgt/hostobjects/APIProviderHostObject.java:3517]: Use Multitenancy Utils to get tenant Id

File: components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/AppMConstants.java [hunk 2]
```diff
@@ -589,4 +589,8 @@
     public static final String PAGE_LOAD_EVENT = "page-load";
 
     public static final int API_AUTH_FORBIDDEN = 900908;
+
+    public static final String ANALYTICS = "Analytics.";
+    public static final String APP_USAGE_BAM_UI_ACTIVITY_ENABLED = ANALYTICS
+            + "UIActivityBAMPublishEnabled";
```
Comment (lakshani) [inline components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/AppMConstants.java:595]: Format the line

File: components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java [hunk 3]
```diff
@@ -6332,166 +6332,153 @@ private boolean isAPIPublished(WebApp api) throws AppManagementException {
...
+        try {
+            conn = APIMgtDBUtil.getUiActivityDBConnection();
+            insertStoreHits(webAppUUID, userId, tenantId, appName, appVersion, context, conn);
             conn.commit();
-		} finally {
-			APIMgtDBUtil.closeAllConnections(null, conn, null);
-		}
-	}
+        } catch (SQLException e) {
+            String webAppInfo =
```
Comment (lakshani) [inline components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:6353]: add rollback

File: components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java [hunk 4]
```diff
@@ -6332,166 +6332,153 @@ private boolean isAPIPublished(WebApp api) throws AppManagementException {
...
+     * @param userId     User Id.
+     * @param tenantId   Tenant Id.
+     * @param appName    App Name.
+     * @param appVersion App Version.
+     * @param context Context.
      */
-    private static void insertUpdateStoreHits(String webAppUUID, String userId,
-                                             Integer maxCount, Integer tenantId, Connection conn)
-            throws AppManagementException {
+    public static void insertStoreHits(String webAppUUID, String userId, Integer tenantId,
```
Comment (lakshani) [inline components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:6373]: change to private

File: components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java [hunk 5]
```diff
@@ -6332,166 +6332,153 @@ private boolean isAPIPublished(WebApp api) throws AppManagementException {
...
             }
-		} catch (SQLException e) {
-            String strDataContext = "(Webapp UUID: " + webAppUUID + ", UserId : " + userId + ", TenantId : " +
-                    tenantId + ", Hit count : " + maxCount + ")";
-            handleException("Error in updating Store hit count: "
-                    + strDataContext + " : " + e.getMessage(), e);
+        } catch (SQLException e) {
+            String webAppInfo = "(Web app UUID: " + webAppUUID + ", UserId : " + userId + ", " +
+                    "TenantId : " + tenantId + ")";
+            handleException("Error occurred while updating Store hits: " + webAppInfo + " : "
```
Comment (lakshani) [inline components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:6396]: Change error propagation

Comment (lahiru823) [inline components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:6396]: do not need to catch. re throw the sql exception

File: components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java [hunk 6]
```diff
@@ -6332,166 +6332,153 @@ private boolean isAPIPublished(WebApp api) throws AppManagementException {
...
-        PreparedStatement ps = null;
-        ResultSet rs = null;
-        String strResult = "";
-
-        // build the final result format string
-        StringBuilder builderResult = new StringBuilder();
-
-        List uuidList = new ArrayList();
+        Connection conn;
+        JSONArray jsonResultArr = null;
```
Comment (lakshani) [inline components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:6413]: jsonResultArr >> jsonResultArray

File: components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java [hunk 7]
```diff
@@ -6332,166 +6332,153 @@ private boolean isAPIPublished(WebApp api) throws AppManagementException {
...
-                        + " LEFT JOIN APM_APP APP ON APP.UUID=HIT.UUID WHERE HIT.USER_ID=? "
-                        + " UNION ALL "
-                        + " SELECT UUID ,0 AS HIT_COUNT, UPPER(APP_NAME) AS APP_NAME FROM APM_APP "
-                        +
-                        " WHERE UUID NOT IN (SELECT UUID FROM APM_APP_HIT_TOTAL WHERE USER_ID=? ) "
-                        + " ORDER BY HIT_COUNT DESC,APP_NAME ASC LIMIT ? , ? ";
+                uuidsList = getAppHitStatsFromAppmDBOnly(conn, userId, startIndex, pageSize,
+                                                         builderDataContext);
+            }
+            jsonResultArr = new JSONArray();
```
Comment (lakshani) [inline components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:6432]: Remove Json Array and put List or map

File: components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java [hunk 8]
```diff
@@ -6332,166 +6332,153 @@ private boolean isAPIPublished(WebApp api) throws AppManagementException {
...
+            for (int i = 0; i < uuidsList.size(); i++) {
+                JSONObject uuidObject = new JSONObject();
+                uuidObject.put("UUID", uuidsList.get(i));
+                jsonResultArr.add(uuidObject);
+            }
+        } catch (SQLException e) {
+            throw new AppManagementException(
```
Comment (lakshani) [inline components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:6439]: Throw SQL Exception.

File: components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java [hunk 9]
```diff
@@ -6500,32 +6487,130 @@ public static List getAppsByHitCount(String userId,
...
+     * @return a list of UUIDs.
+     * @throws AppManagementException
+     */
+    private static List<String> getAppHitStatsFromBamDBAndAppmDB(Connection appMCon, String userId,
+                                                                 int startIndex, int pageSize,
+                                                                 StringBuilder builderDataContext)
+            throws AppManagementException {
+        List<String> uuidsList = new ArrayList<String>();
+        ResultSet bamResultSet = null;
+        PreparedStatement bamPs = null;
```
Comment (lakshani) [inline components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:6523]: bamPs >> bamPreparedStatement
bamConn >> bamDbConnection
appmPs >> appmPreparedStatement

File: components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java [hunk 10]
```diff
@@ -6500,32 +6487,130 @@ public static List getAppsByHitCount(String userId,
...
+                            + builderDataContext.toString() + " : " + e.getMessage(), e);
+        } finally {
+            APIMgtDBUtil.closeAllConnections(bamPs, bamConn, bamResultSet);
+            APIMgtDBUtil.closeAllConnections(appmPs, appMCon, appmResultSet);
+        }
+        return uuidsList;
+    }
+
+    private static String getAppUuidsFromBamDb(Connection bamConn)
+            throws  AppManagementException {
```
Comment (lakshani) [inline components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:6565]: Change method name,
getAppUuidsFromBamDb >> getDbQueryToRetrieveUuidsFromBamDb

File: components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java [hunk 11]
```diff
@@ -6500,32 +6487,130 @@ public static List getAppsByHitCount(String userId,
...
+            }
+        } catch (SQLException ex) {
+            throw new AppManagementException(
+                    "SQL Exception is occurred while reading driver name of BAM database connection "
+                            + " : " + ex.getMessage(), ex);
+        }
+        return uuidRetrivealBamQuery;
+    }
+
+    private static String getAppUuidsFromAppmDb(List<String> uuidList, Connection appmConn)
```
Comment (lakshani) [inline components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:6587]: getAppUuidsFromAppmDb >> getDbQueryToRetrieveUuidsFromAppmDb

File: components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java [hunk 12]
```diff
@@ -6500,32 +6487,130 @@ public static List getAppsByHitCount(String userId,
...
+            }
+            appmPs.setInt(uuidsList.size() + 1, startIndex);
+            appmPs.setInt(uuidsList.size() + 2, pageSize);
+            appmResultSet = appmPs.executeQuery();
+            while (appmResultSet.next()) {
+                uuidsList.add(appmResultSet.getString("UUID"));
+            }
+        } catch (SQLException e) {
+            throw new AppManagementException(
+                    "SQL Exception is occurred while fetching the store hit sorted data from " +
```
Comment (lakshani) [inline components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:6554]: Throw SqlException

File: components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java [hunk 13]
```diff
@@ -6500,32 +6487,130 @@ public static List getAppsByHitCount(String userId,
...
+                            "App Manager Database and Bam database: "
+                            + builderDataContext.toString() + " : " + e.getMessage(), e);
+        } finally {
+            APIMgtDBUtil.closeAllConnections(bamPs, bamConn, bamResultSet);
+            APIMgtDBUtil.closeAllConnections(appmPs, appMCon, appmResultSet);
```
Comment (lakshani) [inline components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:6559]: close appMCon from the parent method

File: components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java [hunk 14]
```diff
@@ -6500,32 +6487,130 @@ public static List getAppsByHitCount(String userId,
...
+     * Get app hits when ui data activity data source has set to BAM data source.
+     * @param appMCon Connection to App Manager database.
+     * @param userId user Id.
+     * @param pageSize No of elements per page.
+     * @param pageSize No of elements per page.
+     * @param builderDataContext builder with data context.
+     * @return a list of UUIDs.
+     * @throws AppManagementException
+     */
+    private static List<String> getAppHitStatsFromBamDBAndAppmDB(Connection appMCon, String userId,
```
Comment (lakshani) [inline components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:6517]: getAppHitStatsFromBamDBAndAppmDB >> getAppHitStatsFromStatsDB

File: components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java [hunk 15]
```diff
@@ -6500,32 +6487,130 @@ public static List getAppsByHitCount(String userId,
...
         } finally {
             APIMgtDBUtil.closeAllConnections(ps, conn, rs);
         }
-        return uuidList;
+        return uuidsList;
     }
 
-	/**
+    /**
+     * Get app hits when ui data activity data source has set to BAM data source.
```
Comment (lakshani) [inline components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:6508]: Add a  descriptive comment

File: components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java [hunk 16]
```diff
@@ -6332,166 +6332,153 @@ private boolean isAPIPublished(WebApp api) throws AppManagementException {
...
+     * Get app hits when ui data activity data source hasn't set to BAM data source.
+     * @param conn Connection.
+     * @param userId user Id.
+     * @param startIndex pagination start index.
+     * @param pageSize No of elements per page.
+     * @param builderDataContext builder with data context.
+     * @return a list of UUIDs.
+     * @throws AppManagementException
+     */
+    private static List<String> getAppHitStatsFromAppmDBOnly(Connection conn, String userId,
```
Comment (lakshani) [inline components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:6457]: getAppHitStatsFromAppmDBOnly >> getAppHitStatsFromAppmDb

File: components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java [hunk 17]
```diff
@@ -6500,32 +6487,130 @@ public static List getAppsByHitCount(String userId,
             ps.setInt(3, startIndex);
             ps.setInt(4, pageSize);
             rs = ps.executeQuery();
-
-            while (rs.next()) {
-                uuidList.add(rs.getString("UUID"));
+            if (rs.isFirst()) {
```
Comment (lakshani) [inline components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:6490]: No need to check condition to initialize the array list

File: components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java [hunk 18]
```diff
@@ -6500,32 +6487,130 @@ public static List getAppsByHitCount(String userId,
...
-
         } catch (SQLException e) {
-            // Exceptions is only logged here (but not thrown externally)
-            // as this method is used only to sort the assets by usage, An
-            // exception in this method should not effect/block the users other
-            // actions
-            log.error(
-                    "SQL Exception while fetching the store hit sorted data : "
-                            + builderDataContext.toString() + " : "
+            throw new AppManagementException(
```
Comment (lakshani) [inline components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:6497]: throw sql exception

File: components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java [hunk 19]
```diff
@@ -6500,32 +6487,130 @@ public static List getAppsByHitCount(String userId,
...
+                    "SQL Exception is occurred while fetching the store hit sorted data from " +
+                            "App Manager Database: " + builderDataContext.toString() + " : "
                             + e.getMessage(), e);
         } finally {
             APIMgtDBUtil.closeAllConnections(ps, conn, rs);
```
Comment (lakshani) [inline components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:6502]: close connection from parent method

File: components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java [hunk 20]
```diff
@@ -6500,32 +6487,130 @@ public static List getAppsByHitCount(String userId,
...
+            while (bamResultSet.next()) {
+                String uuid = bamResultSet.getString("UUID");
+                uuidsList.add(uuid);
+            }
+            String uuidRetrievalAppmQuery = getAppUuidsFromAppmDb(uuidsList,appMCon);
+            appmPs = appMCon.prepareStatement(uuidRetrievalAppmQuery);
+            for (int j = 0; j < uuidsList.size(); j++) {
+                appmPs.setString(j + 1, uuidsList.get(j));
+            }
+            appmPs.setInt(uuidsList.size() + 1, startIndex);
```
Comment (lakshani) [inline components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:6546]: Put a comment to explain why using "uuidsList.size() + 1"

File: components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java [hunk 21]
```diff
@@ -6500,32 +6487,130 @@ public static List getAppsByHitCount(String userId,
...
+
+            if (bamConn.getMetaData().getDriverName().contains("Oracle")) {
+                uuidRetrivealBamQuery = "SELECT UUID, COUNT(*) AS HIT_COUNT, UPPER(APP_NAME) AS "
+                        + "APP_NAME, CONTEXT FROM APM_APP_HITS WHERE USER_ID=? "
+                        + "GROUP BY UUID "
+                        + "WHERE ROWNUM >= ? AND ROWNUM <= ? "
+                        + "ORDER BY HIT_COUNT DESC,APP_NAME ASC";
+            }
+        } catch (SQLException ex) {
+            throw new AppManagementException(
```
Comment (lakshani) [inline components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:6580]: throw sql exception

File: components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java [hunk 22]
```diff
@@ -6500,32 +6487,130 @@ public static List getAppsByHitCount(String userId,
...
+            uuidRetrievalQuery = uuidRetrievalAppmQuery.substring(0, uuidRetrievalAppmQuery.length() - 1);
+
+            if (appmConn.getMetaData().getDriverName().contains("Oracle")) {
+                uuidRetrievalQuery += ")  WHERE ROWNUM >= ? AND ROWNUM <= ? "
+                        + "ORDER BY APP_NAME ASC";
+            } else {
+                uuidRetrievalQuery += ") ORDER BY APP_NAME ASC LIMIT ? , ?";
+            }
+        } catch (SQLException ex) {
+            throw new AppManagementException(
```
Comment (lakshani) [inline components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:6606]: throw sql exception

File: components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java [hunk 23]
```diff
@@ -6500,32 +6487,130 @@ public static List getAppsByHitCount(String userId,
...
+            throws AppManagementException {
+        StringBuilder uuidRetrievalAppmQuery = new StringBuilder();
+        String uuidRetrievalQuery;
+        try {
+            uuidRetrievalAppmQuery.append("SELECT UUID, UPPER(APP_NAME) AS APP_NAME, CONTEXT FROM "
+                                                  + "APM_APP WHERE UUID NOT IN (");
+            for (int i = 0; i < uuidList.size(); i++) {
+                uuidRetrievalAppmQuery.append("?,");
+            }
+            uuidRetrievalQuery = uuidRetrievalAppmQuery.substring(0, uuidRetrievalAppmQuery.length() - 1);
```
Comment (lakshani) [inline components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:6597]: check for empty uuidlist

File: components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java [hunk 24]
```diff
@@ -7443,4 +7528,36 @@ public static String getDisplayOrderSeqNo() throws AppManagementException {
...
 		return  seqNo;
 	}
+
+    public static WebAppInfoDTO getWebAppByTrackingCode(String trackingCode) throws AppManagementException {
+        Connection conn = null;
+        PreparedStatement ps = null;
+        ResultSet rs = null;
+        WebAppInfoDTO webAppInfoDTO = new WebAppInfoDTO();
+
+        String ssoInfoSqlQuery = "SELECT APP_NAME, APP_VERSION, APP_PROVIDER, CONTEXT, " +
```
Comment (lakshani) [inline components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:7538]: format the query

File: components/org.wso2.carbon.appmgt.usage/org.wso2.carbon.appmgt.usage.client/src/main/java/org/wso2/carbon/appmgt/usage/client/APIUsageStatisticsClient.java [hunk 25]
```diff
@@ -381,6 +382,71 @@ public static void initializeDataSource() throws APIMgtUsageQueryServiceClientEx
...
+                                                                     "APIUsageTracker configuration.");
+        }
+
+        Map<String, AppHitsStatsDTO> appHitsStatsMap = new HashMap<String, AppHitsStatsDTO>();
+        Connection connection = null;
+        PreparedStatement getAppHitsStatement = null;
+        ResultSet appInfoResult = null;
+        List<AppHitsStatsDTO> appHitsStatsList = null;
+        try {
+            connection = dataSource.getConnection();
```
Comment (lakshani) [inline components/org.wso2.carbon.appmgt.usage/org.wso2.carbon.appmgt.usage.client/src/main/java/org/wso2/carbon/appmgt/usage/client/APIUsageStatisticsClient.java:408]: put comment to describe the functionality

File: components/org.wso2.carbon.appmgt.usage/org.wso2.carbon.appmgt.usage.client/src/main/java/org/wso2/carbon/appmgt/usage/client/APIUsageStatisticsClient.java [hunk 26]
```diff
@@ -1393,7 +1463,44 @@ private OMElement queryBetweenTwoDaysForAPIUsageByUser(String fromDate, String t
...
+                userHitsPerApp.setContext(context);
+                AppHitsStatsDTO appHitsStats = appHitsStatsMap.get(context);
+                List<UserHitsPerAppDTO> userHitsStatsList = appHitsStats.getUserHitsList();
+                userHitsStatsList.add(userHitsPerApp);
+            }
+            List<AppHitsStatsDTO> appHitsStatsList =
+                    new ArrayList<AppHitsStatsDTO>(appHitsStatsMap.values());
+            return appHitsStatsList;
+        } catch (SQLException e) {
+            throw new APIMgtUsageQueryServiceClientException("SQL Exception is occurred when " +
```
Comment (lakshani) [inline components/org.wso2.carbon.appmgt.usage/org.wso2.carbon.appmgt.usage.client/src/main/java/org/wso2/carbon/appmgt/usage/client/APIUsageStatisticsClient.java:1496]: return sql exception

File: components/org.wso2.carbon.appmgt.usage/org.wso2.carbon.appmgt.usage.client/src/main/java/org/wso2/carbon/appmgt/usage/client/APIUsageStatisticsClient.java [hunk 27]
```diff
@@ -1393,7 +1463,44 @@ private OMElement queryBetweenTwoDaysForAPIUsageByUser(String fromDate, String t
...
+                                            "reading user hits from SQL table" + e.getMessage(), e);
+        } finally {
+            APIMgtDBUtil.closeAllConnections(getAppHitsStatement, null, appInfoResult);
+        }
+    }
+
+    //cache stat page query
 
     private OMElement queryForCacheHitCount(String fromDate, String toDate, Integer limit)
```
Comment (lakshani) [inline components/org.wso2.carbon.appmgt.usage/org.wso2.carbon.appmgt.usage.client/src/main/java/org/wso2/carbon/appmgt/usage/client/APIUsageStatisticsClient.java:1505]: Remove unused queryForCacheHitCount

File: components/org.wso2.carbon.appmgt.usage/org.wso2.carbon.appmgt.usage.client/src/main/java/org/wso2/carbon/appmgt/usage/client/dto/AppHitsStatsDTO.java [hunk 28]
```diff
@@ -0,0 +1,100 @@
...
+* under the License.
+*/
+package org.wso2.carbon.appmgt.usage.client.dto;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Total App Hits with user app stats list.
+ */
```
Comment (lakshani) [inline components/org.wso2.carbon.appmgt.usage/org.wso2.carbon.appmgt.usage.client/src/main/java/org/wso2/carbon/appmgt/usage/client/dto/AppHitsStatsDTO.java:25]: put a descriptive class comment

**Pull request merged**